src/borg/_hashindex.c

```diff
@@ -436,14 +447,18 @@ hashindex_set(HashIndex *index, const void *key, const void *value)
     uint8_t *bucket_ptr;
     int other_offset;
     int entry_size = (index->key_size + index->value_size);
-    static void *entry_to_insert = NULL;
-    static void *tmp_entry = NULL;
-    /* uint8_t entry_to_insert[index->key_size + index->value_size]; */
+    /* void *entry_to_insert = index->entry_to_insert; */
+    /* void *tmp_entry = index->tmp_entry; */
+
+    /* static void *entry_to_insert = NULL; */
+    /* static void *tmp_entry = NULL; */
+    uint8_t entry_to_insert[entry_size];
+    uint8_t tmp_entry[entry_size];
     int num_buckets = index->num_buckets;
-    if (entry_to_insert == NULL) {
-        entry_to_insert = malloc(entry_size * 2);
-        tmp_entry = entry_to_insert + entry_size;
-    }
+    /* if (entry_to_insert == NULL) { */
+    /*     entry_to_insert = malloc(entry_size * 2); */
+    /*     tmp_entry = entry_to_insert + entry_size; */
+    /* } */
```

Comment:
guess there should be some rebase -i at the end, collapsing some comment-back-and-forth changes (and have them replaced by precompiler conditionals in the end).